### PR TITLE
#64 Testing startups: Post, Delete, Put (  sub-task  #13)

### LINKED ISSUE #64: Testing startups: Post, Delete, Put
task #13

### PR DESCRIPTION
**Pull request by DariyBautista**
Tests for creating, editing, deleting (checking for open projects), data validation, filtering and searching startups

### CODE REVIEW

File: ForumProject/startups/tests.py
Comment (mehalyna) [inline ForumProject/startups/tests.py:1]: Consider adding tests that cover more edge cases, such as attempting to update or delete startups that do not exist or providing invalid data formats

File: ForumProject/startups/tests.py [hunk 1]
```diff
@@ -163,4 +166,324 @@ def test_fail_field_mandatory_attr_empty(self):
...
+            
+    def test_create_startup(self):
+        '''
+        The function tests the creation of a new startup through the API.
+        '''
+        url = reverse('startups:startup-add')
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        response = self.client.post(url, self.startup_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Startup.objects.count(), 1)
```
Comment (mehalyna) [inline ForumProject/startups/tests.py:179]: Consider also checking the content of the response for POST, PUT, and GET requests to ensure that the API returns the expected data. This can help catch serialization and permission issues.
E.g.
```
def test_create_startup(self):
    response = self.client.post(self.startup_add_url, self.startup_data, format='json')
    self.assertEqual(response.status_code, status.HTTP_201_CREATED)
    self.assertEqual(Startup.objects.count(), 1)
    self.assertEqual(response.data['startup_name'], 'Django Dribblers') 

```

File: ForumProject/startups/views.py [hunk 2]
```diff
@@ -139,3 +144,37 @@ class StartupListDetailfilter(generics.ListAPIView):
...
+        """
+        Get the queryset of startups created by the current user.
+
+        Returns:
+            QuerySet: Queryset of startups created by the current user.
+
+        """
+        user_id = self.request.user.id
+        # Filter startups created by the current user
+        return Startup.objects.filter(userstartup__customuser=user_id)
```
Comment (mehalyna) [inline ForumProject/startups/views.py]: If the list of startups can grow significantly, consider implementing pagination to manage the data volume efficiently

**Review by mehalyna — APPROVED** [review]

File: ForumProject/startups/views.py (hunk 2 above)
Comment (mehalyna) [inline ForumProject/startups/views.py]: _Note_

**Review by mehalyna — CHANGES_REQUESTED** [review]

File: ForumProject/startups/views.py (hunk 2 above)
Comment (DariyBautista) [reply ForumProject/startups/views.py]: Done, thanks!

Comment (DariyBautista) [reply ForumProject/startups/views.py]: Done, thanks!

File: ForumProject/startups/tests.py (hunk 1 above)
Comment (DariyBautista) [reply ForumProject/startups/tests.py:179]: Many were already done, I added more where needed, thanks!

File: ForumProject/startups/tests.py
Comment (DariyBautista) [reply ForumProject/startups/tests.py:1]: Added, thanks!

**Review by mehalyna — APPROVED** [review]
